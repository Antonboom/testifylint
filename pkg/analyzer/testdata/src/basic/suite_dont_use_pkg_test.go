package basic

import (
	"testing"

	"github.com/stretchr/testify/assert"
	"github.com/stretchr/testify/require"
	"github.com/stretchr/testify/suite"
)

type SuiteDontUsePackageSuite struct {
	suite.Suite
}

func TestSuiteDontUsePackageSuite(t *testing.T) {
	suite.Run(t, new(SuiteDontUsePackageSuite))
}

func (s *SuiteDontUsePackageSuite) TestAll() {
	var b bool

	assert.True(s.T(), b)                         // want "suite-dont-use-pkg: use s\\.True"
	assert.True(s.T(), b, "msg")                  // want "suite-dont-use-pkg: use s\\.True"
	assert.True(s.T(), b, "msg with arg %d", 42)  // want "suite-dont-use-pkg: use s\\.True"
	assert.Truef(s.T(), b, "msg")                 // want "suite-dont-use-pkg: use s\\.Truef"
	assert.Truef(s.T(), b, "msg with arg %d", 42) // want "suite-dont-use-pkg: use s\\.Truef"

	require.False(s.T(), b)                         // want "suite-dont-use-pkg: use s\\.Require\\(\\)\\.False"
	require.False(s.T(), b, "msg")                  // want "suite-dont-use-pkg: use s\\.Require\\(\\)\\.False"
	require.False(s.T(), b, "msg with arg %d", 42)  // want "suite-dont-use-pkg: use s\\.Require\\(\\)\\.False"
	require.Falsef(s.T(), b, "msg")                 // want "suite-dont-use-pkg: use s\\.Require\\(\\)\\.Falsef"
	require.Falsef(s.T(), b, "msg with arg %d", 42) // want "suite-dont-use-pkg: use s\\.Require\\(\\)\\.Falsef"

	// Negative cases are covered by neighbour files.
}
