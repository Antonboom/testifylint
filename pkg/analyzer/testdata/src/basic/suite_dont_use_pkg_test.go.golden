package basic

import (
	"testing"

	"github.com/stretchr/testify/assert"
	"github.com/stretchr/testify/require"
	"github.com/stretchr/testify/suite"
)

type SuiteDontUsePackageSuite struct {
	suite.Suite
}

func TestSuiteDontUsePackageSuite(t *testing.T) {
	suite.Run(t, new(SuiteDontUsePackageSuite))
}

func (s *SuiteDontUsePackageSuite) TestAll() {
	var b bool

	s.True(b)                         // want "suite-dont-use-pkg: use s\\.True"
	s.True(b, "msg")                  // want "suite-dont-use-pkg: use s\\.True"
	s.True(b, "msg with arg %d", 42)  // want "suite-dont-use-pkg: use s\\.True"
	s.Truef(b, "msg")                 // want "suite-dont-use-pkg: use s\\.Truef"
	s.Truef(b, "msg with arg %d", 42) // want "suite-dont-use-pkg: use s\\.Truef"

	s.Require().False(b)                         // want "suite-dont-use-pkg: use s\\.Require\\(\\)\\.False"
	s.Require().False(b, "msg")                  // want "suite-dont-use-pkg: use s\\.Require\\(\\)\\.False"
	s.Require().False(b, "msg with arg %d", 42)  // want "suite-dont-use-pkg: use s\\.Require\\(\\)\\.False"
	s.Require().Falsef(b, "msg")                 // want "suite-dont-use-pkg: use s\\.Require\\(\\)\\.Falsef"
	s.Require().Falsef(b, "msg with arg %d", 42) // want "suite-dont-use-pkg: use s\\.Require\\(\\)\\.Falsef"

	// Negative cases are covered by neighbour files.
}
